FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : ScalarValue<string>
+    {
+        /// <summary>
+        /// Create a Ref from a string, such as <c>new Ref("databases/prydain")</c>.
+        /// </summary>
+        /// <param name="value">Value.</param>
+        public RefV(string value) : base(value)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        protected internal override void WriteJson(JsonWriter writer) =>
+            writer.WriteObject("@ref", Value);
+    }
+
+    public abstract class ObjectRef : RefV
```

Comment:
The point of the abstraction in core is to reduce the necessary exposure to the ref concept early on, hence the motivation for adding the query lang syntax sugar for creating schema objects, and schema refs. Internally everything is still a ref.

I think a good parallel is ruby's metaprogramming API vs its more traditional syntax for defining classes in source. Everything being an object is great, but it's a little mind-bending for daily use.

How that applies to the driver API, perhaps is still open, but they probably should mirror the query API functions as closely as possible.